modules/components/src/Aggs/DatesAgg.js

```diff
@@ -16,55 +16,61 @@ const momentToBucketDate = moment => moment?.format(BUCKET_DATE_FORMAT);
 class DatesAgg extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {
-      ...this.getInitialRange(props),
-      focusedInput: null, // must be set with this.setInputFocus to bind with dom state
-    };
+    this.state = this.initializeState(props);
   }
 
   componentWillReceiveProps(nextProps) {
-    this.setState(this.getInitialRange(nextProps));
+    this.setState(this.initializeState(nextProps));
   }
 
-  getInitialRange = ({ buckets = [], getActiveValue = () => ({}) }) => {
+  initializeState = ({ buckets = [], getActiveValue = () => ({}) }) => {
     const { field } = this.props;
     const bucketMoments = buckets.map(x => bucketDateToMoment(x.key_as_string));
-    const minDate = minBy(bucketMoments, moment => moment.valueOf());
-    const maxDate = maxBy(bucketMoments, moment => moment.valueOf());
+    const minDate = minBy(bucketMoments, x => x.valueOf()).subtract(1, 'days');
+    const maxDate = maxBy(bucketMoments, x => x.valueOf()).add(1, 'days');
```

Comment:
a null check here? `__missing__` as a bucket would cause a breakage on the chained Moment method call.